Tests/ClientTests/Helpers/NotificationManagerTests.swift

```diff
@@ -0,0 +1,103 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+@testable import Client
+
+class NotificationManagerTests: XCTestCase {
+    private var center: MockUserNotificationCenter!
+    private var notificationManager: NotificationManager!
+
+    override func setUp() {
+        super.setUp()
+        center = MockUserNotificationCenter()
+        notificationManager = NotificationManager(center: center)
+    }
+
+    func testRequestAuthorization() {
+        let expectation = self.expectation(description: "notification manager")
+        notificationManager.requestAuthorization { (granted, error) in
+            XCTAssertTrue(granted)
+            XCTAssertTrue(self.center.requestAuthorizationWasCalled)
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 5)
+    }
+
+    func testGetNotificationSettings() {
+        let expectation = self.expectation(description: "notification manager")
+        notificationManager.getNotificationSettings { settings in
+            XCTAssertTrue(self.center.getSettingsWasCalled)
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 5)
+    }
+
+    func testScheduleDate() {
+        notificationManager.schedule(title: "Title",
+                                     body: "Body",
+                                     id: "test-id",
+                                     date: Date.tomorrow)
+        XCTAssertTrue(center.addWasCalled)
+    }
+
+    func testScheduleInterval() {
+        notificationManager.schedule(title: "Title",
+                                     body: "Body",
+                                     id: "test-id",
+                                     interval: 50)
+        XCTAssertTrue(center.addWasCalled)
+    }
+
+    func testFindDeliveredNotifications() {
+        let expectation = self.expectation(description: "notification manager")
+        notificationManager.findDeliveredNotifications { notifications in
+            XCTAssertTrue(self.center.getDeliveredWasCalled)
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 5)
```

Comment:
Since your mock `MockUserNotificationCenter` is synchronous, technically you wouldn't need expectations here I believe.